### PR TITLE
[428] Change terraform vars to json

### PR DESCRIPTION
**Pull request by saliceti**
## Context
As json is a standard format, it allows us to query the values outside of terraform using `jq` for example. This will allow simplifying the configuration and removing Github secrets.

## Changes proposed in this pull request
- Change terraform tfvars to json
- Read keyvault config from tfvars.json instead of secrets
- Extra: deploy the main branch when IMAGE_TAG is not specified

## Guidance to review
Run a manual deployment via make

## Link to Trello card
https://trello.com/c/CHMLdUw6

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by saliceti** [comment]
[Change terraform vars to json](https://trello.com/c/CHMLdUw6/428-change-terraform-vars-to-json)

File: Makefile [hunk 1]
```diff
@@ -102,21 +102,28 @@ research:
 azure-login:
 	az account set -s $(AZURE_SUBSCRIPTION)
 
+read-keyvault-config:
+	$(eval key_vault_name=$(shell jq -r '.key_vault_name' terraform/workspace_variables/$(APP_ENV).tfvars.json))
```
Comment (cwrw) [inline Makefile:106]: I was wondering why those weren't capitilized to be consistent with the variables defined above?

**Review by cwrw — APPROVED** [review]
Looks good 👍  a few questions 
Maybe we should trigger a test deploy to QA with this branch just to make sure all is in order before merging?

File: Makefile [hunk 2]
```diff
@@ -124,7 +124,7 @@ shell: ## Open a shell on the app instance on PaaS, eg: make qa shell
 	cf ssh apply-clock-${APP_NAME_SUFFIX} -t -c 'cd /app && /usr/local/bin/bundle exec rails c'
 
 deploy-init:
-	$(if $(IMAGE_TAG), , $(error Please pass a valid docker image tag; eg: make qa deploy-init IMAGE_TAG=5309326123bf6b366deab6cd0668615d11be3e3d))
+	$(if $(IMAGE_TAG), , $(eval export IMAGE_TAG=main))
```
[the remark this answers is not in the document]
Comment (saliceti) [reply Makefile:134]: Thanks. Added an echo to show the deployed image tag

File: Makefile (hunk 1 above)
Comment (saliceti) [reply Makefile]: Thanks, it looks better indeed

**Review by cwrw — APPROVED** [review]
looks great, I just tested the deploy to qa and 👍

**Pull request merged**